tests/test_errors.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from notion_client import AsyncClient, Client
+from notion_client.errors import (
+    APIResponseError,
+    HTTPResponseError,
+    RequestTimeoutError,
+    is_api_error_code,
+)
+
+STATUS_PAGE_BAD_REQUEST = "https://httpstat.us/400"
+STATUS_PAGE_TIMEOUT = "https://httpstat.us/200?sleep=100"
+
+
+@pytest.mark.vcr()
+def test_api_response_error(client):
+    with pytest.raises(APIResponseError):
+        client.request("/invalid", "GET")
+    with pytest.raises(APIResponseError):
+        client.request("/users", "GET", auth="Invalid")
+
+
+def test_api_request_timeout_error(token):
```

Comment:
No cassette here?

Reply:
So, that's a tricky one. The test makes a request but there is no response as I artificially create a timeout, so the cassette is not created. I don't like it because if there is no connection the test will fail, let me know if there is any workaround

Reply:
If there is no connection then it's going to timeout as well I guess! That probably good enough.